src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,110 @@
+package duke;
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.util.ArrayList;
+
+/**
+ * Class that encapsulates the file reading/writing for the storage of tasks.
+ */
+public class Storage {
+    private File file;
+
+    /**
+     * Class Constructor that takes in the file path of the storage file.
+     *
+     * @param filePath where the storage file is at.
+     */
+    public Storage (String filePath) {
+        File temp = new File(filePath);
+        try {
+            temp.getParentFile().mkdir();
+            temp.createNewFile();
+            this.file = temp;
+        } catch (Exception e) {
+            System.out.println("Can't create file");
+        }
+    }
+
+    /**
+     * Reads the file of the storage and returns a ArrayList of Tasks from the storage.
+     *
+     * @return ArrayList of Task that is stored.
+     */
+    public ArrayList<Task> parseFile () {
+        ArrayList<Task> history = new ArrayList<Task>();
+        try {
+            FileReader fileReader = new FileReader(file);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            String currentLine;
+            while ((currentLine = bufferedReader.readLine()) != null) {
+                Task temp = parseTask(currentLine);
+                history.add(temp);
+            }
+        } catch (Exception e) {
+            System.out.println("File is not found");
+        }
+        return history;
+    }
+
+    /**
+     * Function that parses the line that is stored in the storage file.
+     *
+     * @param task line in the file that represents a task.
+     * @return Task object corresponding to the line in the file.
+     */
+    public Task parseTask(String task) {
+        String[] temp = task.split(" \\| ");
+        if (temp.length == 3) {
```

Comment:
Interesting way to find for Todo. Why not do the same as the Event? (temp[0].equalsIgnoreCase("T")) It may be better as you may be adding tasks in the future whose temp.length == 3 as well.